Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -70,26 +76,42 @@ void doChargeDepositionShapeN(const GetParticlePosition& GetPosition,
             // x direction
             // Get particle position in grid coordinates
 #if (defined WARPX_DIM_RZ)
-            const amrex::Real r = std::sqrt(xp*xp + yp*yp);
-            const amrex::Real x = (r - xmin)*dxi;
+            const amrex::Real rp = std::sqrt(xp*xp + yp*yp);
+            const amrex::Real x = (rp - xmin)*dxi;
 #else
             const amrex::Real x = (xp - xmin)*dxi;
 #endif
-            // Compute shape factors for node-centered quantities
+
+            // Compute shape factor along x
+            // i: leftmost grid point that the particle touches
             amrex::Real sx[depos_order + 1];
-            // i: leftmost grid point (node-centered) that the particle touches
-            const int i = compute_shape_factor<depos_order>(sx,  x);
+            int i;
+            if (rho_type[0] == NODE) {
+                i = compute_shape_factor<depos_order>(sx, x);
+            } else if (rho_type[0] == CELL) {
+                i = compute_shape_factor<depos_order>(sx, x - 0.5);
+            }
```

Comment:
For compactness and to allow `i` to be `const`, we could also do.
```suggestion
            const int i = (rho_type[0] == NODE) ? compute_shape_factor<depos_order>(sx, x)
                                                : compute_shape_factor<depos_order>(sx, x - 0.5);
```
This also applies to y an z.

Actually, we could also have something like
```c++
    i = compute_shape_factor<depos_order>(sx, x, rho_type[0]);
```
where `compute_shape_factor` would take care of the staggering, this could be a bit more intuitive. What do you think? It's just a suggestion, though, we can keep it this way for this PR anyway.

Reply:
Adding the type to the call to compute_shape_factor is a good idea. Though this should be a separate PR since it also affects the current deposition, which uses the same routine.

Reply:
Indeed, I totally agree, thanks!